micrometer-observation/src/main/java/io/micrometer/observation/docs/DocumentedObservation.java

```diff
@@ -187,6 +188,30 @@ default <T extends Observation.Context> Observation observation(@Nullable Observ
         return observation;
     }
 
+    /**
+     * Creates an {@link Observation} for the given {@link ObservationConvention}. You
+     * need to manually start it. The {@link Observation.Context} creation happens after
+     * the {@link ObservationRegistry} type check. When the {@link ObservationRegistry} is
+     * a no-op, this method fast returns a no-op {@link Observation} and skips the
+     * creation of the {@link Observation.Context}.
+     * @param customConvention convention that (if not {@code null}) will override any
+     * pre-configured conventions
+     * @param defaultConvention default convention that will be picked if there was
+     * neither custom convention nor a pre-configured one via
+     * {@link ObservationRegistry.ObservationConfig#observationConvention(GlobalObservationConvention[])}
+     * @param contextSupplier observation context supplier
+     * @param registry observation registry
+     * @return observation
+     */
+    default <T extends Observation.Context> Observation createNotStarted(
```

Comment:
Can we also add a null check since the `registry` is `@Nullable`? See:
https://github.com/micrometer-metrics/micrometer/blob/da7c9f3fd3482d6c546aaf9d66dae30841494731/micrometer-observation/src/main/java/io/micrometer/observation/Observation.java#L102-L105

Also, should not we do this in the `Observation` too?

Reply:
`registry` is non-null on this method, so don't need null check here.

I'll add documentation that this no-op registry check is a quick check and further decision(`ObservationPredicate`) would happen after context is created.

Reply:
I'll go to an eye doctor. :D